sdk/log/record.go

```diff
@@ -216,42 +230,54 @@ func (r *Record) AddAttributes(attrs ...log.KeyValue) {
 		rIndex := r.attrIndex()
 		defer putIndex(rIndex)
 
-		// Unique attrs that need to be added to r. This uses the same underlying
-		// array as attrs.
-		//
-		// Note, do not iterate attrs twice by just calling dedup(attrs) here.
```

Comment:
The replacement algorithm is now iterating `attrs` twice. This seems like a regression and very inefficient.

Reply:
```
AddAttributes/Deduplication/Enabled-20           137.1n ±  4%    230.3n ±  5%   +67.92% (p=0.000 n=10)
AddAttributes/NestedDeduplication/Enabled-20     430.1n ±  1%   1010.0n ±  4%  +134.80% (p=0.000 n=10)
```

I will try to work on it.

Reply:
Can you update the description as well?

Reply:
I already  updated the benchmarks :wink: 

But now I also removed the remark about possible improvements.
If there are any then a follow-up PR is the way to go.